### ISSUE TITLE
frame normalisation gives nan in case of long silence 

### ISSUE THREAD

**Issue by mshenron** [philipperemy/deep-speaker#5]
def normalize_frames(m):
    return [(v - np.mean(v)) / np.std(v) for v in m]

as std(v) is zeros in case of long silence

**Comment by philipperemy** [comment]
@mshenron yes I think we need to reconsider this normalisation. In the paper they don't say what they used. If you have any ideas, please contribute :)

**Comment by mshenron** [comment]
normalisation is giving better learning curve just change the normalisation function to

def normalize_frames(m,epsilon=1e-12):
    return [(v - np.mean(v)) / max(np.std(v),epsilon) for v in m]

**Comment by shubhamaryan** [comment]
This paper "Batch Normalization: Accelerating Deep Network Training by Reducing Internal Covariate Shift" suggests to do batch normalization as follows:

epsilon = 1e-12
def normalize_frames(m):
    return [(v - np.mean(v)) / np.sqrt(np.var(v)+epsilon) for v in m]

**Comment by philipperemy** [comment]
@shubhamaryan thanks! Very helpful comments guys! I'll prepare a PR.

**Comment by philipperemy** [comment]
Fixed
